src/state/routing/utils.ts

```diff
@@ -37,88 +55,113 @@ export function computeRoutes(
         throw new Error('Expected both amountIn and amountOut to be present')
       }
 
-      const routeProtocol = getRouteProtocol(route)
+      const isOnlyV2 = isVersionedRoute<V2PoolInRoute>(PoolType.V2Pool, route)
+      const isOnlyV3 = isVersionedRoute<V3PoolInRoute>(PoolType.V3Pool, route)
```

Comment:
Because the type is tied to the first parameter, you shouldn't need to pass it:

```suggestion
      const isOnlyV2 = isVersionedRoute(PoolType.V2Pool, route)
      const isOnlyV3 = isVersionedRoute(PoolType.V3Pool, route)
```

Reply:
it unfortunately gives me another type error if i do that - will leave it alone for now